### ISSUE TITLE
lgbm.train vs sklearn lgbm api

### ISSUE THREAD

**Issue by germayneng** [microsoft/LightGBM#2418]
Hi,

I am trying to replicate some parameters between non sklearn and sklearn api because some external packages use lgbm sklearn format (like eli5). 

1. I cant seem to find class_weight for non sklearn format (i.e lgbm.train). For non sklearn, is the only way to add weights via weight parameter in lgbm.Dataset? 

https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.Dataset.html#lightgbm.Dataset

i know i can just add n_sample weight array to define weights for each sample according to its class. (for lgbm.train method) So where does sklearn's class_weight parameter goes into the default lgbm.train?

**Comment by StrikerRUS** [comment]
@germayneng Hi!

> I cant seem to find class_weight for non sklearn format (i.e lgbm.train). 

`class_weight` param is presented only in sklearn wrapper. Standard API (`lgb.train`) doesn't have this param because it's general-purpose function and do not distinguish between regression, classification and ranking tasks at the parameters level. Also, `class_weight` is transformed into `sample_weight` via sklearn function, so, it will require sklearn installed as a dependency.

> For non sklearn, is the only way to add weights via weight parameter in lgbm.Dataset?

 Yes, indeed.

> So where does sklearn's class_weight parameter goes into the default lgbm.train?

It goes to `weight` param of `lgb.Dataset` class here:

https://github.com/microsoft/LightGBM/blob/023749239ebf390b5d56e08f227a013cbd972eb0/python-package/lightgbm/sklearn.py#L538-L548

`_LGBMComputeSampleWeight` is actually this sklearn function: https://scikit-learn.org/stable/modules/generated/sklearn.utils.class_weight.compute_sample_weight.html.
